### PR TITLE
Push Notifications: update instructions with hint to refresh

### LINKED ISSUE #6223: Push Notifications: Re-allowing permissions doesn't clear the DISABLED warning without page refresh
#### Steps to reproduce
1. Enable Push Notifications in Firefox
2. Disable the notifications permissions from the Lock icon on the site
3. Refresh/load https://wordpress.com/me/notifications
4. Verify push notifications are marked as DISABLED, with instructions to re-allow
5. Follow said instructions

#### What I expected
The warning label to go away

#### What happened instead
I clicked around inside Calypso and went back to /me/notifications and the warning was still there.  A full page refresh was needed to remove it.  Maybe there should just be a note on the instructions saying to refresh?  Not technically necessary to enable notifs, but it would clear up the confusion.

#### Browser / OS version
OSX 10.11.5 Firefox 46.0

#manual-testing

### PR DESCRIPTION
**Pull request by jonathansadowski**
Fixes #6223

Before:

image

After:

image

To test:
* Test via calypso.live or on calypso.localhost using [this workaround for secure origins](https://www.chromium.org/blink/serviceworker/service-worker-faq)
* Change your browser settings to block notification preferences. Click the security icon in the address bar (or the page icon if testing via calypso.localhost) and then set notifications to "Always block on this site".
* Reload the page
* Visit /me/notifications, you should see this: image
* Click "View Instructions to Enable"

### CODE REVIEW

File: client/me/notification-settings/push-notification-settings/index.jsx [hunk 1]
```diff
@@ -250,6 +250,7 @@ const PushNotificationSettings = React.createClass( {
 							) }</p>
 						</div>
 					</div>
+					<Notice className="notification-settings-push-notification-settings__instruction-refresh-notice" showDismiss={ false } text={ this.translate( 'Once you\'ve allowed notifications, you may need to refresh your browser.' ) } />
```
Comment (mtias) [inline client/me/notification-settings/push-notification-settings/index.jsx]: This class should be showing an ESLint warning: `push-notification-settings` should be the prefix.

**Comment by folletto** [comment]
> Once you've allowed notifications, you may need to refresh your browser.

Isn't this possible for us to detect and do a force refresh ourselves?

File: client/me/notification-settings/push-notification-settings/index.jsx (hunk 1 above)
Comment (jonathansadowski) [inline client/me/notification-settings/push-notification-settings/index.jsx]: Thanks, I've updated the className

**Comment by jonathansadowski** [comment]
> Isn't this possible for us to detect and do a force refresh ourselves?

It is not. The reason a refresh is required is because the new permissions aren't visible/available to us. They do not become visible to us until the user refreshes the page after changing the setting. Chrome's native UI points out that a page refresh is required, but Firefox's does not.

**Comment by folletto** [comment]
> It is not. The reason a refresh is required is because the new permissions aren't visible/available to us. They do not become visible to us until the user refreshes the page after changing the setting. Chrome's native UI points out that a page refresh is required, but Firefox's does not.

Thanks. Super annoying. :|

The above seems good. Might be nicer to have a design ad-hoc that matches the rest of the dialog, or something less "evident", like this:

artboard 1

Or a three-steps illustration instead of two steps where the latest one is refresh.

However, I'm ok also with the above. 👍

**Comment by mattsherman** [comment]
LGTM! 👍

**Comment by jonathansadowski** [comment]
Thanks for the feedback / reviews -- we elected to go with the design as-is because we didn't want to make the instructions any less apparent, and because adding a third step begins to become unruly at certain breakpoints

**Comment by folletto** [comment]
Makes sense 👍

**Pull request merged**